File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        if s == '':
            return True
        
        a = list('abcdefghijklmnopqrstuvwxyz0123456789')
        s = s.lower()
        # a = list(a)
        # a.extend(list(b))
        r = ""
        for e in s:
            if e in a:
                r = r + e
        b = r[::-1]
        # print(r,b)
        return r==b

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.replaceAll("[^a-zA-Z0-9]","").toLowerCase();
        for(int l=0,r=s.length()-1;l<r;l++,r--){
            if(s.charAt(l)!=s.charAt(r)) return false;
        }
        return true;
    }

    public boolean __isPalindrome(String s) {
        if (s.isEmpty()) {
        	return true;
        }
        int start = 0;
        int last = s.length() - 1;
        while(start <= last) {
        	char currFirst = s.charAt(start);
        	char currLast = s.charAt(last);
        	if (!Character.isLetterOrDigit(currFirst )) {
        		start++;
        	} else if(!Character.isLetterOrDigit(currLast)) {
        		last--;
        	} else {
        		if (Character.toLowerCase(currFirst) != Character.toLowerCase(currLast)) {
        			return false;
        		}
        		start++;
        		last--;
        	}
        }
        return true;
    }
}